tests/unit/apps/test_auxiliary.py

```diff
@@ -1020,3 +1026,97 @@ def test_mysql_innodb_cluster_upgrade(model):
     upgrade_plan = app.generate_upgrade_plan(target, False)
 
     assert_steps(upgrade_plan, expected_plan)
+
+
+def test_auxiliary_upgrade_by_unit(model):
```

Comment:
It would not be enough to check that `generate_upgrade_plan` was run with None, even we provided list of units?

Reply:
Yeah, I think that is enough